bn_mp_get_i32.c

```diff
@@ -0,0 +1,15 @@
+#include "tommath_private.h"
+#ifdef BN_MP_GET_I32_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+
+/* Return the least significant 32 bits */
+int32_t mp_get_i32(const mp_int *a)
+{
+   uint64_t res = mp_get_mag32(a);
```

Comment:
Shouldn't this be uint32_t ?    I think it should.

Reply:
Ouch!
Thanks!
(But that is in the part of the preprocessor expansion? Or really a C&P error? Need to check)

Reply:
> But that is in the part of the preprocessor expansion?

Yes, it is in the result of `gcc -E`.
So if this PR is not wanted: @minad , please check your macros.

Reply:
I think both, the `uint64_t` and the `(int32_t)-res` are intentional to avoid UB in case of overflow.
I don't think it works but I was wrong once, so I'll keep my voice low ;-)